r2/r2/controllers/api.py

```diff
@@ -795,6 +800,86 @@ def check_api_friend_oauth_scope(self, type_):
+                    if potentialsuccessor._deleted:
+                        badsuccessordescription = "user is [deleted]"
+                    elif potentialsuccessor._spam:
+                        badsuccessordescription = "user is shadowbanned"
+                    sr.remove_moderator(potentialsuccessor)
+                    # Log this
+                    ModAction.create(
+                        sr,
+                        c.user,
+                        'uninvitemoderator',
+                        target=potentialsuccessor,
+                        details='account is not a valid successor',
+                        description=badsuccessordescription
+                    )
+                    continue
+                # if the successor does not have all perms, give it to him/her
+                if not potentialsuccessor_perms['all']:
+                    potentialsuccessor_as_mod.set_permissions(fullperms)
+                    potentialsuccessor_as_mod._commit()
+                    # Log this
+                    ModAction.create(
+                        sr,
+                        c.user,
+                        'setpermissions',
+                        target=potentialsuccessor,
+                        details='set permissions on successor',
+                        description='+all'
+                    )
+                # break the loop as soon as a successor has been properly found/made
+                break
+        elif len(sr.moderator_invites) > 0:
+            # cycle through each invite
+            fullperms = sr.moderator_invite_permission_class.loads('+all')
+            for user_id in sr.moderator_invites:
+                potentialsuccessor = Account._byID(user_id)
+                potentialsuccessor_as_mod = sr.get_moderator_invite(potentialsuccessor)
+                potentialsuccessor_perms = potentialsuccessor_as_mod.get_permissions()
+                # uninvite each next successor if said successor is deleted / marked as spam
+                if potentialsuccessor._deleted or potentialsuccessor._spam:
+                    if potentialsuccessor._deleted:
+                        badsuccessordescription = "user is [deleted]"
+                    elif potentialsuccessor._spam:
+                        badsuccessordescription = "user is shadowbanned"
+                    sr.remove_moderator(potentialsuccessor)
+                    # Log this
+                    ModAction.create(
+                        sr,
+                        c.user,
+                        'uninvitemoderator',
+                        target=potentialsuccessor,
+                        details='account is not a valid successor',
+                        description=badsuccessordescription
+                    )
+                    continue
+                # if the successor does not have all perms, give it to him/her
+                if not potentialsuccessor_perms['all']:
```

Comment:
I didn't use `has_permissions` here nor on the mod cycle because I couldn't get it to work here. Weirdly enough this syntax worked fine.